### DISCUSSION TITLE
[Question]I want to save some data and release some resources before closing the software.

### DISCUSSION THREAD

**Discussion by molihuan** [flet-dev/flet#2911] [Q&A] [answered]
### Question

I tried to find the lifecycle of building software for Fleet, but I couldn't find it. I want to save some data and release some resources before closing the software. What should I do? Please help me, thank you very much.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
- https://flet.dev/docs/controls/page#window_prevent_close
- https://flet.dev/docs/controls/page/#on_app_lifecycle_state_change

**Comment by jlindoso** [comment]
Do you use userControl?

try to use will_unmount event 

https://flet.dev/docs/guides/python/user-controls/
